cmd/init-osm-controller/init-osm-controller.go

```diff
@@ -112,9 +92,18 @@ func main() {
 		return
 	}
 
-	meshConfig := createDefaultMeshConfig()
+	presetMeshConfig, presetConfigErr := configClient.ConfigV1alpha1().MeshConfigs(osmNamespace).Get(context.TODO(), presetMeshConfigName, metav1.GetOptions{})
+	_, meshConfigErr := configClient.ConfigV1alpha1().MeshConfigs(osmNamespace).Get(context.TODO(), meshConfigName, metav1.GetOptions{})
+
+	// If the presetMeshConfig could not be loaded and a default meshConfig doesn't exist, return the error
+	if presetConfigErr != nil && apierrors.IsNotFound(meshConfigErr) {
+		log.Fatal().Err(err).Msgf("Unable to create default meshConfig, as %s could not be found", presetMeshConfigName)
+		return
+	}
+
+	defaultMeshConfig := createDefaultMeshConfig(presetMeshConfig)
 
-	if createdMeshConfig, err := configClient.ConfigV1alpha1().MeshConfigs(osmNamespace).Create(context.TODO(), meshConfig, metav1.CreateOptions{}); err == nil {
+	if createdMeshConfig, err := configClient.ConfigV1alpha1().MeshConfigs(osmNamespace).Create(context.TODO(), defaultMeshConfig, metav1.CreateOptions{}); err == nil {
```

Comment:
I have a concern here. Suppose a scenario:

1. user installed OSM
2. user updated meshconfig, using manifest file and `kubectl apply -f` command.
3. somehow osm-controller crashes and restarted.

In this case, MeshConfig will be reverted to the preset.

We need to call out that preset MeshConfig has higher priority, and will overwrite live MeshConfig during osm controller initialization.

Reply:
This will not happen, if there's an already existing meshconfig in the cluster, it will not get updated from preset ( the create api will only create a new resource if one doesn't exist, it will not override an existing resource). I have verified this by testing it locally. 

You can give it a try and let me know if you have any concerns

Reply:
Steps I ran locally to verify this:
1. I created a mesh config on my cluster before installing osm ( with values different from default).
2.  installed osm, and noticed that the meshconfig was as I installed earlier